src/engine/framework/VirtualMachine.cpp

```diff
@@ -370,7 +370,7 @@ uint32_t VMBase::Create()
 {
 	type = static_cast<vmType_t>(params.vmType.Get());
 
-	if (type < 0 || type >= TYPE_END)
+	if (type < TYPE_NACL || type >= TYPE_END)
```

Comment:
I find this change makes the intent less clear. How about adding a TYPE_BEGIN alias for the first enumerator?

Reply:
Done.

Can you, please, review fixes and added commits?